### ISSUE TITLE
[deploy-sourcegraph] Add support for using Crossplane to provision managed services in Kubernetes deployment

### ISSUE THREAD

**Issue by hasheddan** [sourcegraph/sourcegraph#12493]
When Sourcegraph is deployed on Kubernetes it instantiates a number of data services including Redis and Postgres. The manifests allow for [configuring your own custom data services](https://docs.sourcegraph.com/admin/install/kubernetes/configure#configure-custom-redis), presumably provisioned via a cloud provider console or some IaC tool as a separate step. [Crossplane](https://github.com/crossplane/crossplane) supports provisioning managed services via Kubernetes manifests, so users who wanted to utilize hosted offerings for their data services in their sourcegraph deployment could simply modify a few configuration values and have these hosted services provisioned on a variety of providers.

I would be happy to add support for this to the sourcegraph k8s deployment configuration, as well as document how it can be utilized. Please let me know if this is something the team would be interested in :slightly_smiling_face: 

/cc @beyang

**Comment by beyang** [comment]
This would be a great thing to incorporate into our docs. Assuming we'll cover this in the October livestream, but happy to carry on an async convo here in the meantime (sorry about the delay, I'm terribly behind on GH notifs). I'm not terribly up-to-date with the latest developments in the k8s ecosystem and Crossplane is rather new to me. I see it's created by Upbound.io, but seems to be a standalone OSS project that doesn't depend on Upbound (though I assume there are synergies)? Could you clarify the relationship? This is important because we need to convey clearly to our customers if a particular installation path will require or even nudge them to purchase additional software.

**Comment by hasheddan** [comment]
@beyang certainly! Crossplane is a [CNCF sandbox project](https://www.cncf.io/sandbox-projects/) that was started by Upbound and is a standalone project (you can read more about its neutral governance [here](https://github.com/cncf/toc/blob/master/proposals/sandbox/crossplane.md)). In addition, the TBS livestream show is a Crossplane community project (https://github.com/crossplane/tbs) rather than an Upbound or Upbound Cloud show. We do not present anything on the show that folks cannot do with 100% free and open source software :slightly_smiling_face:
